src/QuadForm/indefiniteLLL.jl

```diff
@@ -0,0 +1,357 @@
+
+################################################################################
+#
+#                       indefinite LLL reduction
+#
+################################################################################
+
+################################################################################
+#                  Helpful
+################################################################################
+
+#=
+  _round_matrix(M::MatElem) -> MatElem
+
+Returns the matrix $M$ where all entries are replaced by their rounded values.
+
+$Example$
+==========
+julia> _round_matrix(QQ[1//3 -2 3//2; 5//3 0 -6; -7//3 -81//50 0])
+[0 -2 2]
+[2 0 -6] 
+[-2 -2 0]
+=#
+function _round_matrix(M::MatElem)
+  return map_entries(round, M)
+end
+
+
+#=
+  _abs_matrix(M::MatElem) -> MatElem
+
+Returns the matrix $M$ where all entries are replaced by their absolute values.
+
+$Example$
+==========
+julia> _abs_matrix(ZZ[1 -2 3; 4 0 -6; -7 -81 0])
+[1 2 3]
+[4 0 6] 
+[7 81 0]
+=#
+function _abs_matrix(M::MatElem)
+    return map_entries(abs, M)
+end
+
+#=
+  vecextract(M::MatElem,x::Int64,y::Int64) -> MatElem
+
+Extracts components of $M$ with regards to the binary representation of $x$ and $y$. 
+The location of the one entries give the position of the entries which should be extracted.
+For example if $x$ = $y$ = 13 = (1101)_2 then M[[1,3,4], [1,3,4]] is returned.
+
+$Example$
+==========
+julia> vecextract(ZZ[1 2 3; 4 5 6; 7 8 9],6,3)
+[4 5]
+[7 8]
+=#
+function _vecextract(M::MatElem,x::Int64,y::Int64)
+  x_bin = digits(x,base = 2)
+  y_bin = digits(y, base = 2)
+
+  list_x = [i for i = 1:length(x_bin) if x_bin[i] == 1]
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[list_x , list_y]
+
+end
+
+function _vecextract(M::MatElem, x::Union{Int64,Vector{Int64}},y::Int64)
+  y_bin = digits(y, base = 2)
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[x , list_y]
+
+end
+
+################################################################################
+#                           linear algebra
+################################################################################
+
+#=
+    _mathnf(A::MatElem) -> MatElem, MatElem
+
+Given a rectangular matrix $A$ of dimension $nxm$. Computes the Hessian matrix $H$ of dimension $nxm$ 
```

Comment:
```suggestion
Given a rectangular matrix $A$ of dimension $nxm$, compute a Hermite normal form $H$ of dimension $nxm$ 
```
Hessian matrices are matrices of second partial derivatives from analysis. Here you are computing a Hermite normal form and you return an unitary matrix `U`. Maybe you could mention that the output is not a regular hnf, but that the non-trivial block and the zero block are switched.